src/DurationInputMask.test.js

```diff
@@ -118,35 +119,50 @@ describe('Handles props', () => {
     expect(ref.focus).toHaveBeenCalled();
   });
 
-  it('value not masked when props.maskOnChange=false', () => {
-    const value = '2m';
-    const nextValue = '62m';
-    const wrapper = shallow(
-      <DurationInputMask value={value} maskOnChange={false} />,
-    );
+  it('event.target.value=undefined handled gracefully', () => {
+    const value = 61;
+    const wrapper = shallow(<DurationInputMask value={value} />);
+    const event = { target: { value: undefined } };
 
-    wrapper.simulate('change', { target: { value: nextValue } });
-    expect(wrapper.state('value')).toEqual(nextValue);
+    wrapper.simulate('change', event);
+
+    expect(wrapper.state('value')).toEqual();
   });
 
-  it('state not updated when props.maskOnChange=false & props.handleChange={fn}', () => {
-    const value = '2m';
+  it('props value is masked on setTimeout', (done) => {
+    const value = 61;
+    const delay = 250;
+    const delayInMilliseconds = delay / 1000;
+    const nextValue = '1m 1s';
     const wrapper = shallow(
-      <DurationInputMask value={value} handleChange={jest.fn()} maskOnChange={false} />,
+      <DurationInputMask maskDelay={delay} value={value} />,
     );
+    const mockHandleMaskDelay = jest.spyOn(wrapper.instance(), 'handleMaskDelay');
 
-    wrapper.simulate('change', { target: { value: '2m 1s' } });
-    expect(wrapper.state('value')).toEqual(value);
+    const before = performance.now();
+
+    wrapper.simulate('change', { target: { value } });
+
+    mockHandleMaskDelay(value, nextValue);
+
+    const after = performance.now();
+
+    expect(wrapper.state('value')).toEqual(nextValue);
+    expect(after - before).toBeGreaterThan(delayInMilliseconds);
+    done();
   });
 
-  it('event.target.value=undefined handled gracefully', () => {
+  it('props value is masked immediately onBlur even if maskDelay is set', () => {
     const value = 61;
-    const wrapper = shallow(<DurationInputMask value={value} />);
-    const event = { target: { value: undefined } };
+    const delay = 1000;
+    const nextValue = '1m 1s';
+    const wrapper = shallow(
+      <DurationInputMask maskDelay={delay} value={value} />,
+    );
 
-    wrapper.simulate('change', event);
+    wrapper.simulate('blur');
 
-    expect(wrapper.state('value')).toEqual('');
+    expect(wrapper.state('value')).toEqual(nextValue);
```

Comment:
I think we should be using `performance.now()` if asserting _when_ a state change happens.

Reply:
Use performance to check the state changed before the maskDelay